sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Rules/BooleanLiteralUnnecessaryBase.cs

```diff
@@ -0,0 +1,272 @@
+            var elseIsBooleanLiteral = IsBooleanLiteral(elseNoParantheses);
+
+            var bothSideBool = thenIsBooleanLiteral && elseIsBooleanLiteral;
+            var bothSideTrue = IsKind(thenNoParantheses, TrueLiteral) && IsKind(elseNoParantheses, TrueLiteral);
+            var bothSideFalse = IsKind(thenNoParantheses, FalseLiteral) && IsKind(elseNoParantheses, FalseLiteral);
+
+            if (bothSideBool && !bothSideFalse && !bothSideTrue)
+            {
+                var location = Location.Create(ternaryTree,
+                    new TextSpan(thenBranch.SpanStart, elseBranch.Span.End - thenBranch.SpanStart));
+
+                context.ReportDiagnosticWhenActive(Diagnostic.Create(SupportedDiagnostics[0], location));
+            }
+            if (thenIsBooleanLiteral ^ elseIsBooleanLiteral)
+            {
+                // one side is boolean literal, the other is NOT boolean literal
+                var booleanLiteralSide = thenIsBooleanLiteral ? thenBranch : elseBranch;
+
+                context.ReportDiagnosticWhenActive(Diagnostic.Create(SupportedDiagnostics[0], booleanLiteralSide.GetLocation()));
+            }
+        }
+
+        protected bool CheckForNullabilityAndBooleanConstantsReport(TBinaryExpression binaryExpression,
+            SyntaxNodeAnalysisContext context, bool reportOnTrue)
+        {
+            var binaryExpressionLeft = GetLeftNode(binaryExpression);
+            var binaryExpressionRight = GetRightNode(binaryExpression);
+
+            var typeLeft = context.SemanticModel.GetTypeInfo(binaryExpressionLeft).Type;
+            var typeRight = context.SemanticModel.GetTypeInfo(binaryExpressionRight).Type;
+            if (ShouldNotReport(typeLeft, typeRight))
+            {
+                return true;
+            }
+
+            var leftIsTrue = IsKind(binaryExpressionLeft, TrueLiteral);
+            var leftIsFalse = IsKind(binaryExpressionLeft, FalseLiteral);
+            var rightIsTrue = IsKind(binaryExpressionRight, TrueLiteral);
+            var rightIsFalse = IsKind(binaryExpressionRight, FalseLiteral);
+
+            var leftIsBoolean = leftIsTrue || leftIsFalse;
+            var rightIsBoolean = rightIsTrue || rightIsFalse;
+
+            if (leftIsBoolean && rightIsBoolean)
+            {
+                var bothAreSame = (leftIsTrue && rightIsTrue) || (leftIsFalse && rightIsFalse);
+                var errorLocation = bothAreSame
+                    ? CalculateExtendedLocation(binaryExpression, false)
+                    : CalculateExtendedLocation(binaryExpression, reportOnTrue == leftIsTrue);
+
+                context.ReportDiagnosticWhenActive(Diagnostic.Create(SupportedDiagnostics[0], errorLocation));
+                return true;
+            }
+            return false;
+        }
+
+        protected static bool ShouldNotReport(ITypeSymbol typeLeft, ITypeSymbol typeRight) =>
+            typeLeft == null
```

Comment:
This coding style is not consistent with the one used on the following method. I'd personally recommend the operator on the new line but no indentation.